File: a_memory-address/stack_heap.c
#include <stdio.h>
#include <stdlib.h>

int global_var = 42; // global/static memory

int main() {
    int stack_var = 10; // stack variable
    int *heap_var = malloc(sizeof(int)); // heap allocation
    *heap_var = 20;

    printf("Address of stack_var: %p\n", (void*)&stack_var);
    printf("Address of heap_var: %p\n", (void*)heap_var);
    printf("Address of global_var: %p\n", (void*)&global_var);

    free(heap_var); // always free heap memory
    return 0;
}

File: a_memory-address/stack_heap.rs
fn main() {
    let stack_var = 10;              // stack variable
    let heap_var = Box::new(20);     // heap allocation using Box
    static GLOBAL_VAR: i32 = 42;     // global/static memory

    println!("Address of stack_var: {:p}", &stack_var);
    println!("Address of heap_var: {:p}", &*heap_var); // deref Box
    println!("Address of GLOBAL_VAR: {:p}", &GLOBAL_VAR);
}